packages/cli/src/util/extension/exec.ts

```diff
@@ -0,0 +1,90 @@
+import which from 'which';
+import execa from 'execa';
+import { dirname } from 'path';
+import { listen } from 'async-listen';
+import { scanParentDirs, walkParentDirs } from '@vercel/build-utils';
+import { createProxy } from './proxy';
+import type Client from '../client';
+
+/**
+ * Attempts to execute a Vercel CLI Extension.
+ *
+ * If the extension was found and executed, then the
+ * exit code is returned.
+ *
+ * If the program could not be found, then an `ENOENT`
+ * error is thrown.
+ */
+export async function execExtension(
+  client: Client,
+  name: string,
+  args: string[],
+  cwd: string
+): Promise<number> {
+  const { debug } = client.output;
+  const extensionCommand = `vercel-${name}`;
+
+  const { packageJsonPath, lockfilePath } = await scanParentDirs(cwd);
+  const baseFile = lockfilePath || packageJsonPath;
+  let extensionPath: string | null = null;
+
+  if (baseFile) {
+    // Scan `node_modules/.bin` works for npm / pnpm / yarn v1
+    // TOOD: add support for Yarn PnP
+    extensionPath = await walkParentDirs({
+      base: dirname(baseFile),
+      start: cwd,
+      filename: `node_modules/.bin/${extensionCommand}`,
+    });
+  }
+
+  if (!extensionPath) {
+    // Attempt global `$PATH` lookup
+    extensionPath = which.sync(extensionCommand, { nothrow: true });
```

Comment:
Why sync?

Reply:
I didn't see any benefit to using async in this case. There's no other async working happening at this point that this sync call could block on.

Sync should technically be faster anyways since it avoids the thread pool.

Reply:
I'm fine with it being sync, but your statement that "Sync should technically be faster anyways since it avoids the thread pool" has piqued my curiosity. Do you think other areas that we know the code path is guaranteed to be synchronous that we should use the sync versions of functions?